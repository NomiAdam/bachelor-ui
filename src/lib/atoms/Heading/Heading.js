// eslint-disable-next-line no-unused-vars
import React, { createElement } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { resolveTheme, TYPES } from '../../utils/resolveTheme';

const HeadingWrapper = styled.div`
  color: ${ resolveTheme(TYPES.COLOR) };
  cursor: ${ ({ cursor }) => cursor };
`;

const Heading = ({
    component, label, width, cursor, ...others
}) => (
    <HeadingWrapper>
        {createElement(component, others, label)}
    </HeadingWrapper>
);
Heading.propTypes = {
    /**
   * String representation of desired element
   */
    component: PropTypes.string,
    /**
   * String representation of display text
   */
    label: PropTypes.string,
    /**
   * String representation of heading underline
   */
    width: PropTypes.string,
    /**
   * String representation of desired css cursor property
   */
    cursor: PropTypes.string,
};

Heading.defaultProps = {
    component: 'h2',
    cursor: 'initial',
};

export default Heading;
