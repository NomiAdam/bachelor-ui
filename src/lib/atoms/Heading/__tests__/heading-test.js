/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import Heading from '../Heading';

const mockLabel = 'label';

describe('Heading', () => {
    it('should be defined', () => {
        expect(Heading).toBeDefined();
    });
    it('should match the snapshot', () => {
        expect(shallow(<Heading />)).toMatchSnapshot();
    });
    it('should have matching label value inside heading', () => {
        expect(shallow(<Heading label={ mockLabel } />).find('h2').text()).toBe(mockLabel);
    });
});
