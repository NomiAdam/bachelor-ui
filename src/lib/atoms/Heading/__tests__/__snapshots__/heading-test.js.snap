// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading should match the snapshot 1`] = `
<styled.div>
  <h2 />
</styled.div>
`;
